src/KokkosPluginLoader.cpp

```diff
@@ -99,6 +99,8 @@ void KokkosPluginLoader::initPlugin(const std::string &path)
   getFunction<post_function>(plugin, post_functions, "kokkosp_end_parallel_for");
 
   getFunction<finalize_function>(plugin, finalize_functions, "kokkosp_finalize_library");
+
+  dlclose(plugin);
```

Comment:
I think this should be called after finalize to ensure the library isn't unloaded while its being used.

Reply:
Hmm, to do that we may have to add an array of pointers so there's something to dlclose. One thing to note is that finalize isn't called by default in RAJA, the user chooses to run it in their code. Maybe call dlclose in the destructor of KokkosPluginLoader? That might be a little volatile though. Thoughts?

Reply:
That makes sense, but do we delete the plugin objects?
Ya maybe its probably better not to close it if there isn't a safe place to do so.
Does it make sense to unload and remove dynamically loaded plugins after finalize?

Reply:
We do delete the plugin objects later on. The array of plugins in RuntimePluginLoader is implemented as: std::vector<std::unique_ptr<PluginStrategy>> plugins; The default destructors of vector and std::unique_ptr take care of it for us. Also, I do believe that's why we didn't have dlclose now that I remember it. It would indeed make sense to unload the plugins after finalize, might be worth adding in a 'plugins.clear()' at the end of finalize. Wouldn't stop the user from adding more plugins afterwards since the vector would still be there

Reply:
That makes sense, I missed the unique_ptr.
Does the plugin store the dlopened library and dlcose it when its destroyed?